src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php

```diff
@@ -116,14 +116,18 @@ static function (AdapterInterface $tagsAdapter, array $tags) {
      */
     public function invalidateTags(array $tags)
     {
-        $ids = [];
+        $ids = $tagsToUpdate = [];
         foreach ($tags as $tag) {
             \assert('' !== CacheItem::validateKey($tag));
             unset($this->knownTagVersions[$tag]);
             $ids[] = $tag.static::TAGS_PREFIX;
         }
 
-        return !$tags || $this->tags->deleteItems($ids);
+        foreach ($this->tags->getItems($ids) as $tag => $version) {
+            $tagsToUpdate[$tag] = $version->set($newVersion ?? $newVersion = random_int(\PHP_INT_MIN, \PHP_INT_MAX));
```

Comment:
There's an issue with this, which was also existing with the previous strategy but was fixed by always deleting items: unused tags are never removed and will last forever in the DB.